lib/ansible/module_utils/gcp_utils.py

```diff
@@ -129,8 +129,8 @@ def _credentials(self):
             credentials, project_id = google.auth.default()
             return credentials
         elif cred_type == 'serviceaccount':
-            return service_account.Credentials.from_service_account_file(
-                self.module.params['service_account_file'])
+            path = os.path.expanduser(self.module.params['service_account_file'])
```

Comment:
I'm not sure it's necessary, but adding an `os.path.realpath` here wouldn't hurt.